pennylane/_grad.py

```diff
@@ -180,7 +180,12 @@ def _jacobian_function(*args, **kwargs):
         if len(argnum) == 1:
             return _jacobian(func, argnum[0])(*args, **kwargs)
 
-        return np.stack([_jacobian(func, arg)(*args, **kwargs) for arg in argnum]).T
+        jacobians = [_jacobian(func, arg)(*args, **kwargs) for arg in argnum]
+
+        try:
+            return np.stack(jacobians).T
+        except ValueError:
```

Comment:
What is it that might cause a `ValueError`?

Reply:
Oh, this might happen if the Jacobian for each argument is a different shape, in which case `stack` will fail! This is actually super common, e.g.:

```python
def cost(x, y):
    return np.sin(np.sum(x)) + np.cos(y)

x = np.array([0.1, 0.2], requires_grad=True)
y = np.array(0.3, requires_grad=True)

>>> cost(x, y)
1.2508566957869456
>>> qml.jacobian(cost)(x, y)
(array([0.95533649, 0.95533649]), array(-0.29552021))
```

Reply:
```suggestion
        except ValueError:
            # The Jacobian of each argument is a different shape and cannot
            # be stacked; simply return the tuple of argument Jacobians.
```